napalm-ruckus-fastiron/fi.py

```diff
@@ -0,0 +1,1378 @@
+        """
+        Closes the connection to the device.
+        """
+        self.device.disconnect()
+
+    def is_alive(self):
+        """
+        Returns a flag with the connection state.
+        Depends on the nature of API used by each driver.
+        The state does not reflect only on the connection status (when SSH), it must also take into
+        consideration other parameters, e.g.: NETCONF session might not be usable, although the
+        underlying SSH session is still open etc.
+        """
+        null = chr(0)
+        try:
+            self.device.send_command(null)
+        except (socket.error, EOFError):
+            return {
+                'is_alive': False
+            }
+        return {
+            'is_alive': self.device.remote_conn.transport.is_active()
+        }
+
+    def _send_command(self, command):
+        """Wrapper for self.device.send.command().
+
+        If command is a list will iterate through commands until valid command.
+        """
+        output = ""
+
+        try:
+            if isinstance(command, list):
+                for cmd in command:
+                    output = self.device.send_command(cmd)
+                    if "% Invalid" not in output:
+                        break
+            else:
+                output = self.device.send_command(command)
+            return output
+        except (socket.error, EOFError) as e:
+            raise ConnectionClosedException(str(e))
+
+    @staticmethod
+    def retrieve_all_locations(long_string, word, pos):
+        """Finds a word of a long_string and returns the value in the nth position"""
+        count = 0                           # counter
+        split_string = long_string.split()  # breaks long string into string of substring
+        values = []                         # creates a list
+        for m in split_string:              # goes through substrings one by one
+            count += 1                      # increments counter
+            if m == word:                   # if substring and word match then specific value
+                values.append(split_string[count + pos])    # is added to list that is returned
+        return values
+
+    @staticmethod
+    def find_words(output, word_list, pos_list):
+        """   """
+        dictionary = {}
+        if len(word_list) != len(pos_list):             # checks word, pos pair exist
+            return None
+
+        if len(word_list) == 0 or len(pos_list) == 0:   # returns NONE if list is empty
+            return None
+
+        size = len(word_list)
+        sentence = output.split()                   # breaks long string into separate strings
+
+        for m in range(0, size):                    # Iterates through size of word list
+            pos = int(pos_list.pop())               # pops element position and word pair in list
+            word = word_list.pop()
+            if sentence.__contains__(word):         # checks if word is contained in text

```

Comment:
Think all the `__contains__` calls might be more pythonic/readable if rewritten as `if word in sentence:`

Reply:
Maybe not for this PR, but need to start thinking about the test framework -- you can look at https://github.com/napalm-automation-community/napalm-mos/tree/develop/test/unit for a barebones example

Reply:
Great point, I agree much more readable. Will make this change.